packages/applications/request-context/src/sendVerificationRequest.ts

```diff
@@ -0,0 +1,43 @@
+import { EmailUserConfig } from 'next-auth/providers/email';
+
+import { SendEmail } from '@potentiel-applications/notifications';
+import { Option } from '@potentiel-libraries/monads';
+import { Routes } from '@potentiel-applications/routes';
+
+import { GetUtilisateurFromEmail } from './getUtilisateur';
+import { canConnectWithProvider } from './canConnectWithProvider';
+
+type BuildSendVerificationRequest = (
+  sendEmail: SendEmail,
+  getUtilisateurFromEmail: GetUtilisateurFromEmail,
+) => NonNullable<EmailUserConfig['sendVerificationRequest']>;
+
+export const buildSendVerificationRequest: BuildSendVerificationRequest =
+  (sendEmail, getUtilisateurFromEmail) =>
+  async ({ identifier, url }) => {
+    const utilisateur = await getUtilisateurFromEmail(identifier);
+
+    if (Option.isSome(utilisateur) && utilisateur.désactivé) {
+      return;
+    }
+
+    if (Option.isNone(utilisateur) || canConnectWithProvider('email', utilisateur.role.nom)) {
+      return sendEmail({
+        templateId: 6785365,
+        messageSubject: 'Connexion à Potentiel',
+        recipients: [{ email: identifier, fullName: '' }],
+        variables: {
+          url,
+        },
+      });
+    }
+
+    return sendEmail({
```

Comment:
Je crois que la condition fonctionne mais c'est pas méga intuitif qu'il ne reste que cette option

Reply:
Inverser les conditions juste pour avoir le return en fin ;) 

```suggestion
    if (Option.isNone(utilisateur) || canConnectWithProvider('email', utilisateur.role.nom)) {
      return sendEmail({
        templateId: 6785365,
        messageSubject: 'Connexion à Potentiel',
        recipients: [{ email: identifier, fullName: '' }],
        variables: {
          url,
        },
      });
    }

    if (!utilisateur.désactivé) {
      return sendEmail({
        templateId: 7103248,
        messageSubject: 'Potentiel - Connexion avec ProConnect obligatoire',
        recipients: [{ email: identifier, fullName: '' }],
        variables: {
          url: Routes.Auth.signIn({ forceProConnect: true }),
        },
      });
    }

    return;```